### PR TITLE
optimize site and reduce page load

### LINKED ISSUE #1184: Speed up initial load
Currently, the web page loads slower than V1. see https://www.webpagetest.org/video/compare.php?tests=220223_AiDc5G_DDD,220223_BiDcMX_CY6

### PR DESCRIPTION
**Pull request by sheralim012**
Closes #1184 

- optimized fetching banner pages with linked pages
- optimized color scheme fetching for buttons, sections, articles, questionnaires
- optimized fetching of SVG to PNG map
- optimized fetching of translations
- added conn max age environment variable for postgres
- added cached session backend
- added switchable cache backend (db/redis/default)

if you want to use database cache then update `docker-compose.yml` to set 'db' as CACHE_BACKEND and run the following command to create database cache table
```
python manage.py createcachetable
```
if you want to use redis cache then update `docker-compose.yml` and set 'redis' as CACHE_BACKEND

### CODE REVIEW

File: iogt/settings/base.py
Comment (ehmadzubair) [inline iogt/settings/base.py:464]: raise exception in else

Comment (sheralim012) [reply iogt/settings/base.py:465]: done

File: iogt/middleware.py
Comment (sheralim012) [reply iogt/middleware.py]: done

**Comment by geoo89** [comment]
I tested this in dev.

- translations added in the translation manager never got refreshed.
- Page preview is not always instant:
    1. Edit a page
    2. Preview (the changes are visible)
    3. Edit the page again
    4. Preview again (the new changes are not visible and take a while to propagate)
- When a user posts a comment, and they are redirected to the current page, their comment is not shown until the next cache refresh. This is confusing.
- Same for responding to the chatbot.

Page edits took a while to propagate to the live pages. I assume this is intentional. Some changes however cannot be previewed, e.g.
- Changing the color of a top-level section
- Editing/Adding a nav-bar entry
- Adding/editing footers
As not all pages are refreshed simultaneously, some pages show the changes earlier than others.
Note that global themeing does immediately take effect.

It seems like all but the translation issue are related to the page-level caching. Should we disable this for now and see how much speedup we get without it?

**Comment by ChrisMarsh82** [comment]
Split this into 2 PRs. Separating caching from other changes.
Add option to have Redis, DB, or no cache.

**Comment by sheralim012** [comment]
@geoo89 I have removed page-level caching, could you check now

**Comment by geoo89** [comment]
LGTM
@istride Any concerns from you end?

Please merge in #1197 (or develop once #1197 has been merged into develop) so we can see if the tests are still succeeding.

**Comment by geoo89** [comment]
Sorry, could you merge in the latest develop again? The cypress DB has conflicts. Once that's done, I will merge.

File: iogt/settings/base.py
Comment (istride) [inline iogt/settings/base.py:17]: I don't think this is used anywhere.

File: iogt/settings/base.py
Comment (istride) [inline iogt/settings/base.py:437]: Redis is our preferred cache backend, but my concern is that we may not be able to set up a Redis cluster in the production environment before this PR is merged and released, so a temporary solution _may_ be necessary.

Using `DatabaseCache` as a temporary solution is not very appealing to me because a new cache table needs to be created for every deployment.

Using `LocMemCache` creates an in-memory cache for each process, which could be quite memory-intensive and lead to unpredictable results due to caches not being synchronised.

The `FileBasedCache` may be best as a temporary solution because it only requires a readable and writable directory on the local filesystem, which would be automatically created if it does not exist. I'm assuming that it is shared by all processes and that it's performance would be worse than in-memory, but better than database caching. Would you agree?

File: docker-compose.yml
Comment (istride) [inline docker-compose.yml:27]: We have a redis server available but we're using 'db' as the cache backend?

**Review by istride — CHANGES_REQUESTED** [review]

File: iogt/settings/test.py
Comment (sheralim012) [reply iogt/settings/test.py:13]: didn't put it in `base.py` as it uses SQLite3 by default

File: iogt/settings/base.py
Comment (sheralim012) [reply iogt/settings/base.py:457]: done, 
also simplified this structure by using `CACHE_BACKEND`, `CACHE_LOCATION`, and `CACHE_TIMEOUT` environment variables and avoid using if else

File: iogt/settings/base.py
Comment (sheralim012) [reply iogt/settings/base.py:458]: done

File: iogt/settings/base.py
Comment (sheralim012) [reply iogt/settings/base.py:464]: it is suggested by wagtail to improve image lookup https://docs.wagtail.org/en/stable/advanced_topics/performance.html#caching-image-renditions

File: iogt/middleware.py
Comment (sheralim012) [reply iogt/middleware.py:122]: removed it as added configurable timeout env. var

File: iogt/settings/base.py
Comment (sheralim012) [reply iogt/settings/base.py:17]: yes, you are right, removed it

File: docker-compose.yml
Comment (sheralim012) [reply docker-compose.yml:27]: we can use whichever cache backend we want to use or test here

File: iogt/settings/base.py
Comment (sheralim012) [reply iogt/settings/base.py:437]: in `FileBasesCache` each cache value will be stored as a separate value and the user running the server has to have read and write access for it to work, see [here](https://docs.djangoproject.com/en/4.0/topics/cache/#filesystem-caching)

Comment (istride) [reply iogt/settings/base.py]: After discussion with @ChrisMarsh82, it has been decided to have two options for cache backend:

- Redis
- No cache (bypass)

If necessary, an intermediate option can be implemented later. We should be prepared to release the changes in this PR without Redis infrastructure in place.

Comment (sheralim012) [reply iogt/settings/base.py]: if we don't specify any cache django by default uses Local-Memory Cache, see [here](https://docs.djangoproject.com/en/4.0/topics/cache/#local-memory-caching-1)

Comment (istride) [reply iogt/settings/base.py]: I would like to have the option to not cache at all, just in case there is any trouble.

File: iogt/settings/test.py
Comment (istride) [reply iogt/settings/test.py:13]: Is there any harm in adding it to `base.py`? I'm assuming that it would be ignored if SQLite is used. My aim is to reduce duplication.

File: iogt/settings/base.py
Comment (sheralim012) [reply iogt/settings/base.py]: let's discuss in a quick call

File: docker-compose.yml
Comment (sheralim012) [reply docker-compose.yml]: if we have decided to go with Redis or no-cache then I believe no-cache would be the default option

File: iogt/settings/test.py
Comment (sheralim012) [reply iogt/settings/test.py:13]: sure, added

**Review by istride — APPROVED** [review]

File: docker-compose.yml
Comment (sheralim012) [reply docker-compose.yml]: used dummy cache as default option

**Pull request merged**
